alea/test/compare.cpp

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright (C) 1998-2018 ALPS Collaboration. See COPYRIGHT.TXT
+ * All rights reserved. Use is subject to license terms. See LICENSE.TXT
+ * For use in publications, see ACKNOWLEDGE.TXT
+ */
+#include <alps/alea.hpp>
+#include <alps/alea/util/model.hpp>
+#include <alps/testing/near.hpp>
+
+#include <iostream>
+#include <random>
+#include <Eigen/Dense>
+
+#include "gtest/gtest.h"
+
+using namespace alps;
+
+template <typename Acc>
+class eq_case
+    : public ::testing::Test
+{
+private:
+    alea::util::var1_model<double> model_;
+
+public:
+    eq_case()
+    {
+        Eigen::VectorXd phi0(2), veps(2);
+        Eigen::MatrixXd phi1(2,2);
+        phi0 << 2, 3;
+        phi1 << .90, 0, 0, .30;
+        veps << 1.0, 0.25;
+        model_ = alea::util::var1_model<double>(phi0, phi1, veps);
+    }
+
+    void fill(Acc &acc, size_t init, size_t n)
+    {
+        alea::util::var1_run<double> run = model_.start();
+        std::mt19937 rng(init);
+        for (; run.t() < 100; run.step(rng))
+            ;
+        for (; run.t() < 100 + n; run.step(rng))
+            acc << run.xt();
+    }
+
+    void test_empty()
+    {
+        Acc acc1(2), acc2;
+        acc2 = acc1;
+        EXPECT_EQ(acc1.result(), acc2.result());
+    }
+
+    void test_one()
+    {
+        Acc acc1(2), acc2(2);
+
+        fill(acc1, 0, 1);
+        EXPECT_NE(acc1.result(), acc2.result());
+
+        acc2 = acc1;
+        EXPECT_EQ(acc1.result(), acc2.result());
```

Comment:
We do rely here on the fact that assignment operator makes objects equal.
Or, in other words, "equal objects are, _by definition_, those which are made by an assignment operator".

Markus noted that IEEE FP standard does not even guarantee that two results made from identical accumulators will be strictly numerically equal --- but if they are indeed not equal on some specific platform and this test fails, we would like to know about it.

Reply:
Good point, thanks for making it explicit.
The test is mainly supposed to take of two corner cases:
 - 0 data points, where the mean does not exist
 - 1 data point, where the variance does not exist